starter_code/src/movies.js

```diff
@@ -1,19 +1,68 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
 
- 
-// Iteration 2: Drama movies - Get the average of Drama Movies
-
-
-// Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
-
-
-// Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
-
-
-// Iteration 5: Alphabetic Order - Order by title and print the first 20 titles
-
-// Iteration 6: Time Format - Turn duration of the movies from hours to minutes
-
-// BONUS Iteration: Best yearly rate average - Best yearly rate average
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+//
+function ratesAverage(moviesArray) {
+    let sum = moviesArray.reduce(function(accum, currentValue) {
+       return accum + currentValue.rate;
+    }, 0);
+    const averageRate = (sum / moviesArray.length).toFixed(2);
+    return parseFloat(averageRate);
+  }
+  // Iteration 2: Drama movies - Get the average of Drama Movies
+  function dramaMoviesRate(moviesArray) {
+      let sum = moviesArray.reduce(function(accum, currentValue) {
+         return accum + currentValue.rate;
+      }, 0);
+      const averageRate = (sum / moviesArray.length).toFixed(2);
+      return parseFloat(averageRate);
+    }
+  // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+  function orderByYear(moviesArray) {
+    let orderedMovies = moviesArray.sort(function(a, b){
+      if (a.year !== b.year) {
+        return a.year - b.year;
+      }
+        return a.title.localeCompare(b.title);
+    });
+    return orderedMovies;
+  }
+  // Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
+  function howManyMovies(moviesArray) {
```

Comment:
At the moment you are just sorting through the movies array. You want to filter all the drama movies by Steven S. 

e.g.
function howManyMovies(movies) {
  let dramaMoviesSteven = movies.filter(function(movie) {
    if (movie.director === "Steven Spielberg" && movie.genre.includes("Drama"))
      return true;
  });
  return dramaMoviesSteven.length;
}